src/MizarWidgets/SamplingWithFrameTrackInputWidgetTest.cpp

```diff
@@ -52,6 +50,27 @@ class MockPairedData {
               (const absl::flat_hash_set<TID>&, FrameTrackId, RelativeTimeNs, RelativeTimeNs),
               (const));
 };
+
+static constexpr size_t kFrameTracksCount = 4;
+static constexpr std::array<FrameTrackId, kFrameTracksCount> kScopeIds = {
+    FrameTrackId(ScopeId(2)), FrameTrackId(ScopeId(1)), FrameTrackId(ScopeId(10)),
+    FrameTrackId(PresentEvent::kD3d9)};
+
+class MockFrameTrackListModel : public QAbstractListModel {
+ public:
+  MockFrameTrackListModel(const MockPairedData*, const absl::flat_hash_set<TID>*,
+                          const RelativeTimeNs*, QObject*) {}
+
+  [[nodiscard]] int rowCount(const QModelIndex& /*parent*/) const override {
+    return kFrameTracksCount;
```

Comment:
You still need to check whether `parent` is valid. If it's valid then you have to return 0, telling the view that a particular list entry has no children.

Reply:
As discussed offline, this should be fixed in production implementations of models. The mock is fine as is.